cdap-security/src/main/java/io/cdap/cdap/security/server/AbstractLoginService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.server;
+
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+/** AbstractLoginService
+ * AbstractLoginService to provide support for username case conversion
+ */
+public abstract class AbstractLoginService extends AbstractLifeCycle implements LoginService {
+
+    private static final Logger LOG = Log.getLogger(AbstractLoginService.class);
+    protected CConfiguration cConfiguration;
+    protected String usernameCaseConversion = "";
+    protected char[] usernameCamelCaseConversionDelimiterArr = {'.'};
+
+    protected AbstractLoginService() {
+        cConfiguration = CConfiguration.create();
+
+        String usernameCaseConversion =
+                cConfiguration.get(Constants.Security.SECURITY_AUTH_USERNAME_CASE_CONVERSION_TYPE);
+        if (usernameCaseConversion != null) {
+            this.usernameCaseConversion = usernameCaseConversion.toLowerCase();
+        }
+
+        String usernameCamelCaseConversionDelimiters =
+                cConfiguration.get(Constants.Security.SECURITY_AUTH_USERNAME_TITLECASE_DELIMITER);
+        if (usernameCamelCaseConversionDelimiters != null && usernameCamelCaseConversionDelimiters.length() > 0) {
+            this.usernameCamelCaseConversionDelimiterArr = usernameCamelCaseConversionDelimiters.toCharArray();
+        }
+
+    }
+
+    public String usernameCaseConvert(String username) {
```

Comment:
Use `@Nullable` for parameter and return that can be null.

Reply:
Also, seems like this should be a `protected` method.

Reply:
yup protected, done